src/webgpu/shader/execution/memory_layout.spec.ts

```diff
@@ -785,6 +785,142 @@ const kLayoutCases: Record<string, LayoutCase> = {
+  array_mat3x2h_stride: {
+    type: `array<mat3x2h, 2>`,
+    read_assign: `out = u32(in[1][0][0])`,
+    write_assign: `out[1][0][0] = f16(in)`,
+    offset: 12,
+    f16: true,
+    skip_uniform: true,
+  },
+  array_mat4x2f_stride: {
+    type: `array<mat4x2f, 4>`,
+    read_assign: `out = u32(in[1][0][0])`,
+    write_assign: `out[1][0][0] = f32(in)`,
+    offset: 32,
+    f32: true,
+  },
+  array_mat4x2h_stride: {
+    type: `array<mat4x2h, 2>`,
+    read_assign: `out = u32(in[1][0][0])`,
+    write_assign: `out[1][0][0] = f16(in)`,
+    offset: 16,
+    f16: true,
+    skip_uniform: true,
+  },
+  array_mat2x3f_stride: {
+    type: `array<mat2x3f, 4>`,
+    read_assign: `out = u32(in[1][0][0])`,
+    write_assign: `out[1][0][0] = f32(in)`,
+    offset: 32,
+    f32: true,
+  },
+  array_mat2x3h_stride: {
+    type: `array<mat2x3h, 2>`,
+    read_assign: `out = u32(in[1][0][0])`,
+    write_assign: `out[1][0][0] = f16(in)`,
+    offset: 16,
+    f16: true,
+    skip_uniform: true,
+  },
+  array_mat3x3f_stride: {
+    type: `array<mat3x3f, 3>`,
+    read_assign: `out = u32(in[2][0][0])`,
+    write_assign: `out[2][0][0] = f32(in)`,
+    offset: 96,
+    f32: true,
+  },
+  array_mat3x3h_stride: {
+    type: `array<mat3x3h, 2>`,
+    read_assign: `out = u32(in[1][0][0])`,
+    write_assign: `out[1][0][0] = f16(in)`,
+    offset: 24,
+    f16: true,
+    skip_uniform: true,
+  },
+  array_mat4x3f_stride: {
+    type: `array<mat4x3f, 4>`,
+    read_assign: `out = u32(in[1][0][0])`,
+    write_assign: `out[1][0][0] = f32(in)`,
+    offset: 64,
+    f32: true,
+  },
+  array_mat4x3h_stride: {
+    type: `array<mat4x3h, 2>`,
+    read_assign: `out = u32(in[1][0][0])`,
+    write_assign: `out[1][0][0] = f16(in)`,
+    offset: 32,
+    f16: true,
+    skip_uniform: true,
+  },
+  array_mat2x4f_stride: {
+    type: `array<mat2x4f, 4>`,
+    read_assign: `out = u32(in[1][0][0])`,
+    write_assign: `out[1][0][0] = f32(in)`,
+    offset: 32,
+    f32: true,
+  },
+  array_mat2x4h_stride: {
+    type: `array<mat2x4h, 2>`,
+    read_assign: `out = u32(in[1][0][0])`,
+    write_assign: `out[1][0][0] = f16(in)`,
+    offset: 16,
+    f16: true,
+    skip_uniform: true,
+  },
+  array_mat3x4f_stride: {
+    type: `array<mat3x4f, 3>`,
+    read_assign: `out = u32(in[2][0][0])`,
+    write_assign: `out[2][0][0] = f32(in)`,
+    offset: 96,
+    f32: true,
+  },
+  array_mat3x4h_stride: {
+    type: `array<mat3x4h, 2>`,
+    read_assign: `out = u32(in[1][0][0])`,
+    write_assign: `out[1][0][0] = f16(in)`,
+    offset: 24,
+    f16: true,
+    skip_uniform: true,
+  },
+  array_mat4x4f_stride: {
+    type: `array<mat4x4f, 4>`,
+    read_assign: `out = u32(in[1][0][0])`,
+    write_assign: `out[1][0][0] = f32(in)`,
+    offset: 64,
+    f32: true,
+  },
+  array_mat4x4h_stride: {
+    type: `array<mat4x4h, 2>`,
+    read_assign: `out = u32(in[1][0][0])`,
+    write_assign: `out[1][0][0] = f16(in)`,
+    offset: 32,
+    f16: true,
+    skip_uniform: true,
```

Comment:
Lots of these new test cases have `skip_uniform: true`, but the array strides looks like they should satisfy the 16-byte requirement for `uniform`?

Reply:
Good catch, I've minimized the skips.